numpy/_core/_add_newdocs_scalars.py

```diff
@@ -369,6 +369,22 @@ def add_newdoc_for_scalar_type(obj, fixed_aliases, doc):
         False
         """))
 
+    add_newdoc('numpy._core.numerictypes', float_name, ('__trunc__',
+        f"""
+        {float_name}.__trunc__() -> int
+
+        Return the floating point number with the fractional part removed,
+        leaving the integer part.
+        {'\n        .. versionadded:: 2.3\n' if float_name != 'double' else ''}
+        Examples
+        --------
+        >>> np.{float_name}(np.pi).__trunc__()
+        3
+        >>> import math
+        >>> math.trunc(np.{float_name}(-np.pi))
+        -3
+        """))
```

Comment:
a classic `.format` might be a bit cleaner here, also for the triple `{float_name}`